syntax = "proto3";

package connection;

message SubscribeRequest{
    string topic = 1;
    string client = 2;
}

message SubscribeResponse{
    bool subsResponse = 1;
}

message PostRequest{
    string text = 1;
    string topic = 2;
    string publisher = 3;
}

message PostResponse{
    bool postedResponse = 1;
    string textResponse = 2;
}

message LoginRequest{
    string username = 1;
}

message LoginResponse{
    repeated string topics = 1;
    int32 idNumber = 2;
}

service SubscribeService{
    rpc SubcribeToTopic(SubscribeRequest) returns (SubscribeResponse){}
}

service PostIntoTopicService{
    rpc PostIntoTopic(PostRequest) returns (PostResponse){}
}

service LoginService{
    rpc LoginIntoApp(LoginRequest) returns (LoginResponse){}
}